homeassistant/components/ezviz/siren.py

```diff
@@ -0,0 +1,115 @@
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EZVIZ sensors based on a config entry."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities(
+        EzvizSirenEntity(coordinator, camera, SIREN_ENTITY_TYPE)
+        for camera in coordinator.data
+        for capability, value in coordinator.data[camera]["supportExt"].items()
+        if capability == str(SupportExt.SupportActiveDefense.value)
+        if value != "0"
+    )
+
+
+class EzvizSirenEntity(EzvizEntity, SirenEntity, RestoreEntity):
+    """Representation of a EZVIZ Siren entity."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = SirenEntityFeature.TURN_ON | SirenEntityFeature.TURN_OFF
+
+    def __init__(
+        self,
+        coordinator: EzvizDataUpdateCoordinator,
+        serial: str,
+        description: SirenEntityDescription,
+    ) -> None:
+        """Initialize the Siren."""
+        super().__init__(coordinator, serial)
+        self._attr_unique_id = f"{serial}_{description.key}"
+        self.entity_description = description
+        self._attr_is_on = False
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        if not (last_state := await self.async_get_last_state()):
+            return
+        self._attr_is_on = last_state.state == STATE_ON
+
+        if self._attr_is_on:
+            evt.async_call_later(self.hass, OFF_DELAY, self.off_delay_listener)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off camera siren."""
+        try:
+            if await self.hass.async_add_executor_job(
+                self.coordinator.ezviz_client.sound_alarm, self._serial, 1
+            ):
+                self._attr_is_on = False
+                self.async_write_ha_state()
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Failed to turn siren off for {self.name}"
+            ) from err
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on camera siren."""
+        try:
+            if await self.hass.async_add_executor_job(
+                self.coordinator.ezviz_client.sound_alarm, self._serial, 2
+            ):
+                self._attr_is_on = True
+                evt.async_call_later(self.hass, OFF_DELAY, self.off_delay_listener)
+                self.async_write_ha_state()
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Failed to turn siren on for {self.name}"
+            ) from err
+
+    @callback
+    def off_delay_listener(self, now: datetime) -> None:
+        """Switch device off after a delay.
+
+        Camera firmware has hard coded turn off after 60 seconds.
+        """
+        self._attr_is_on = False
+        self.async_write_ha_state()
```

Comment:
Should we not instead refresh the coordinator and update the state from the data instead of just assuming the device is off?

Reply:
Unfortunately, EZVIZ doesn't have an api to get the actual state. (Even the app just uses a timer). 

I'm sure they will release something in the future like they did with the alarm panel.

Reply:
But then if we update only here the coordinator would think it's still on right and then change the state back again?

Reply:
Still need a reply here thanks

Reply:
The state is handled in memory only.  Just using the API to turn the siren on, it then also turns off after exactly 60 seconds even if the camera loses connectivity to the network.

It's also safe to trigger twice, the camera just ignores the 2nd trigger if the siren is already active.

Reply:
Right, so you should
- Remove the coordinator as it's not used for this entity
- Set polling to off (already off by the coordinator but since you remove it, you need to set it)
- Set assumed state to true as we are assuming a state from now on

Reply:
Use EzvizBaseEntity, which doesn't inherit the supervisor. (Still need the coordinator for api calls and it's session cookie)
Set polling to disabled.
The assumed state attribute, when set, doesn't indicate status of the entity (show both on/off buttons). The state is set initially and on every change (in memory) so this shouldn't be required I think?

Reply:
I see I'm wrong when it comes to assumed state, it does show status with both on/off buttons. I'll fix.

Reply:
- Set assumed state flag to true.